presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -895,7 +896,9 @@ public FragmentProperties setDistribution(PartitioningHandle distribution, Metad
             }
 
             if (currentPartitioning.equals(SOURCE_DISTRIBUTION)) {
-                this.partitioningHandle = Optional.of(distribution);
+                if (!isPreferLocalUnion(session)) {
```

Comment:
The purpose of `PlanFragmenter::setDistribution` is to (1) set PartitioningHandle for this fragment and  (2) resolve `PartitioningHandle` conflict when multiple table scans exist, e.g., partial_merge_push_down

I cannot think of a case when this logics (a SOURCE_DISTRIBUTION is overwritten to new `distribution`) will happen. I comment this line out in https://github.com/prestodb/presto/pull/14993, all tests pass.